File: Demo/Swift/项目/图文混排11.14/图文混排11.14/ViewController.m
//
//  ViewController.m
//  图文混排11.14
//
//  Created by WZH on 16/11/14.
//  Copyright © 2016年 limpid. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *label;

@end

@implementation ViewController

//目标: 我[爱你]
- (void)viewDidLoad {
    [super viewDidLoad];
   
    //Attachment - 附件
    NSTextAttachment *attachment = [[NSTextAttachment alloc] init];
    
    attachment.image = [UIImage imageNamed:@"d_aini"];
    //提示 lineHeight 大致和字体的大小相等
    CGFloat height = self.label.font.lineHeight;
    attachment.bounds = CGRectMake(0, -4, height, height);
    
    /**
     frame 和 bounds 区别
     
     frame:x,y 决定当前控件相对于父控件的位置
     bounds: x,y 决定内部子控件相对于远点的位置，就是scrollview的contentOffset
     
     
     */
    //2.图像字符串
    NSAttributedString *imageStr = [NSAttributedString attributedStringWithAttachment:attachment];
    
    //3.定义一个可变的属性字符串
    NSMutableAttributedString *attrStrM = [[NSMutableAttributedString alloc] initWithString:@"我"];
    
    //4.拼接图片文本
    [attrStrM appendAttributedString:imageStr];
    
    //设置属性文本
    self.label.attributedText = attrStrM;
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Demo/Swift/语法/27.Swift3.0 懒加载和OC的区别/27.Swift3.0 懒加载和OC的区别/ViewController.swift
//
//  ViewController.swift
//  27.Swift3.0 懒加载和OC的区别
//
//  Created by WZH on 16/10/20.
//  Copyright © 2016年 limpid. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    private lazy var label: UILabel = UILabel()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        label.text = "hello"
        label.sizeToFit()
        
        //和OC 不同
        //一旦 label 被设置为nil，懒加载也不会被再次执行
        //懒加载的代码智慧在第一次调用的时候，执行闭包，然后将闭包的结果保存在label的属性中
        print(label)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        
        //Swift中一定注意不要主动的清理视图或者空间
        //因为懒加载不会再次创建 
    }
}

